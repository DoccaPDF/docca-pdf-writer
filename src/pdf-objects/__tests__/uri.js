import URI from '../uri';

const uri = URI({ id: 42, Base: 'http://docca.io' });

describe('pdf-objects', () => {
  describe('uri', () => {
    it('creates a new object', () => {
      expect(uri).toMatchSnapshot();
    });

    it('flattens keys values', () => {
      expect(uri.flattenKeys()).toMatchSnapshot();
    });

    it('returns a PDF object', () => {
      expect(uri.toPDF()).toMatchSnapshot();
    });
  });
});
