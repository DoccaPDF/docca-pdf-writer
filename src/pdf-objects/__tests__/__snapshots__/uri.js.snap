// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pdf-objects uri creates a new object 1`] = `
Object {
  "Base": "http://docca.io",
  "id": 42,
}
`;

exports[`pdf-objects uri flattens keys values 1`] = `
Object {
  "Base": "(http://docca.io)",
  "Type": "/URI",
}
`;

exports[`pdf-objects uri returns a PDF object 1`] = `
"42 0 obj
<<
/Type /URI
/Base (http://docca.io)
>>
endobj
"
`;
